manager/src/main/java/org/openremote/manager/mqtt/TeltonikaMQTTHandler.java

```diff
@@ -0,0 +1,627 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.Timestamp;
+import java.text.MessageFormat;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import static org.openremote.manager.event.ClientEventService.CLIENT_INBOUND_QUEUE;
+import static org.openremote.manager.mqtt.DefaultMQTTHandler.prepareHeaders;
+import static org.openremote.model.syslog.SyslogCategory.API;
+import static org.openremote.model.value.MetaItemType.*;
+
+public class TeltonikaMQTTHandler extends MQTTHandler {
+
+    //TODO: Maybe move this to Models?
+    private static class TeltonikaDevice {
+        String clientId;
+        String commandTopic;
+
+        public TeltonikaDevice(Topic topic) {
+            this.clientId = topic.tokens.get(1);
+            this.commandTopic = String.format("%s/%s/teltonika/%s/commands",
+                    topicRealm(topic),
+                    this.clientId,
+                    topic.tokens.get(3));
+        }
+    }
+
+    private static final String TELTONIKA_DEVICE_RECEIVE_TOPIC = "data";
+    private static final String TELTONIKA_DEVICE_SEND_TOPIC = "commands";
+    private static final String TELTONIKA_DEVICE_TOKEN = "teltonika";
+    private static final String TELTONIKA_DEVICE_SEND_COMMAND_ATTRIBUTE_NAME = "sendToDevice";
+    private static final String TELTONIKA_DEVICE_RECEIVE_COMMAND_ATTRIBUTE_NAME = "response";
+
+    private static final Logger LOG = SyslogCategory.getLogger(API, TeltonikaMQTTHandler.class);
+
+    protected AssetStorageService assetStorageService;
+    protected AssetDatapointService AssetDatapointService;
+    protected TimerService timerService;
+    protected Path DeviceParameterPath;
+
+    protected final ConcurrentMap<String, TeltonikaDevice> connectionSubscriberInfoMap = new ConcurrentHashMap<>();
+
+
+    /**
+     * Indicates if this handler will handle the specified topic; independent of whether it is a publish or subscribe.
+     * Should generally check the third token (index 2) onwards unless {@link #handlesTopic} has been overridden.
+     *
+     */
+    @Override
+    protected boolean topicMatches(Topic topic) {
+        return TELTONIKA_DEVICE_TOKEN.equalsIgnoreCase(topicTokenIndexToString(topic, 2));
+    }
+
+    @Override
+    public void start(Container container) throws Exception {
+        super.start(container);
+        getLogger().info("Starting Teltonika MQTT Handler");
+        ManagerIdentityService identityService = container.getService(ManagerIdentityService.class);
+        assetStorageService = container.getService(AssetStorageService.class);
+        AssetDatapointService = container.getService(AssetDatapointService.class);
+        timerService = container.getService(TimerService.class);
+        DeviceParameterPath = Paths.get("deployment/manager/fleet/FMC003.json");
```

Comment:
This could be baked into the resources of the module unless a user needs to change it.

This relative path won't work when image is packaged into docker.

Reply:
Can you show me an example for this so that I can implement it?